### PR TITLE
Tutorial editor

### PR DESCRIPTION
**Pull request by brian-c**
This switches the tutorial from being baked into the project's config to being linked as a separate resource.

There's also a rudimentary editor in place. I'm working with this to figure out a good way to organize components and data with Redux in the future.

The editor can be activated in a project's admin settings.

### CODE REVIEW

File: app/classifier/tutorial-button.cjsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  componentDidMount: ->
+    @fetchTutorialFor @props.project
+
+  componentWillReceiveProps: (nextProps) ->
+    unless nextProps.project is @props.project
+      @fetchTutorialFor nextProps.project
+
+  fetchTutorialFor: (project) ->
+    apiClient.type('tutorials').get project_id: @props.project.id
```
Comment (aweiksnar) [inline app/classifier/tutorial-button.cjsx]: Should `@props.project.id` be `project.id` here?

File: app/components/media-card.cjsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ module.exports = React.createClass
     src: ''
 
   render: ->
+    console.log 'src', @props.src
```
Comment (aweiksnar) [inline app/components/media-card.cjsx]: Extra log

**Comment by brian-c** [comment]
Fixed both.

**Comment by aweiksnar** [comment]
Thanks looks good :+1:  I added a tutorial to one of my dev projects and works great. Not sure how I feel about the _boundForceUpdate pattern, but happy to wait until the upcoming architecture changes on that

**Comment by brian-c** [comment]
I bound `forceUpdate` instead of using a `ChangeListener`. With the current architecture, how would you go about updating a component when a resource changes?

**Comment by aweiksnar** [comment]
Could resource updates happen in the parent and then have a shouldComponentUpdate in the child? 
Proposed client edits could be passed up and the top level would trigger saving the new resource

**Comment by brian-c** [comment]
The Controller is the top level, (aside from the Fetcher, which isn't concerned with anything other than making sure the resource is available).

In redux-speak, I'm making the Controller a "smart component" and everything below it a "dumb component", but because we're not using a global store, the Controller re-renders when its resource changes. With redux there'll be an app-level Provider, which I believe is essentially just a app-level forceUpdate when the store changes, right?

I'm under the impression that `shouldComponentUpdate` is mainly for performance optimization, not really necessary here.

**Comment by aweiksnar** [comment]
Alright, Yeah I guess it's just a style preference then. The docs recommend avoiding forceUpdate , so wanted to bring it up if there was another way

**Pull request merged**